src/Tests/dotnet-run.Tests/GivenDotnetRunRunsCsProj.cs

```diff
@@ -572,5 +572,74 @@ public void ItForwardsEmptyArgumentsToTheApp()
                 .And
                 .HaveStdOutContaining($"0 = a{Environment.NewLine}1 = {Environment.NewLine}2 = c");
         }
+
+        [Fact]
+        public void ItDoesNotPrintBuildingMessageByDefault()
+        {
+            var expectedValue = "Building...";
+            var testAppName = "TestAppSimple";
+            var testInstance = _testAssetsManager.CopyTestAsset(testAppName)
+                .WithSource();
+
+            new DotnetCommand(Log, "run")
+               .WithWorkingDirectory(testInstance.Path)
+               .Execute()
+               .Should()
+               .Pass()
+               .And
+               .NotHaveStdOutContaining(expectedValue);
+        }
+
+        [Fact]
+        public void ItPrintsBuildingMessageIfLaunchSettingHasDotnetRunMessagesSet()
+        {
+            var expectedValue = "Building...";
+            var testAppName = "TestAppWithLaunchSettings";
+            var testInstance = _testAssetsManager.CopyTestAsset(testAppName)
+                .WithSource();
+
+            new DotnetCommand(Log, "run")
+               .WithWorkingDirectory(testInstance.Path)
+               .Execute()
+               .Should()
+               .Pass()
+               .And
+               .HaveStdOutContaining(expectedValue);
+        }
+
+        [Fact]
+        public void ItConcatenatesHostingStartups()
+        {
+            var expectedValue = "ASPNETCORE_HOSTINGSTARTUPASSEMBLIES=HostingStartupAmbient;HostingStartupInLaunchSettings";
+            var testAppName = "TestAppWithLaunchSettings";
+            var testInstance = _testAssetsManager.CopyTestAsset(testAppName)
+                .WithSource();
+
+            new DotnetCommand(Log, "run")
+               .WithWorkingDirectory(testInstance.Path)
+               .WithEnvironmentVariable("ASPNETCORE_HOSTINGSTARTUPASSEMBLIES", "HostingStartupAmbient")
```

Comment:
You are testing if `HostingStartupInLaunchSettings` will be added from `LaunchSettings`? The test name is not very clear.